accounts/models.py

```diff
@@ -39,6 +40,9 @@ def deposit(self, amount: decimal.Decimal) -> None:
         if type(amount) is not decimal.Decimal:
             amount: decimal.Decimal = decimal.Decimal(amount)
 
+        if amount < 0:
```

Comment:
Do you tested this conditional validation between a `decimal.Decimal` and an `int`?

I'm not sure if it works. But anyway, I guess it would be good to convert the integer value to be a decimal value. Just to keep the pattern used along the software.

```suggestion
        if amount < decimal.Decimal(0.0):
```

Reply:
Thanks for the suggestion! And yes, I tested my code and it worked just fine.
That being said, I fixed it anyway since it might be safer to work the way you suggested.